test/docfx.Test/lib/ProcessUtilityTest.cs

```diff
@@ -113,113 +113,99 @@ public static void ParallelNestedRunInMutexWithSameNameTest()
-
-
-        [InlineData(new[] { "a-s:a", "r-s:a", "a-e:a", "r-e:a" }, new[] { true, true, true, true })]
-        [InlineData(new[] { "a-e:a", "r-e:a", "a-s:a", "r-s:a" }, new[] { true, true, true, true })]
-        [InlineData(new[] { "a-s:a", "a-e:a", "r-s:a", "r-e:a" }, new[] { true, false, true, false })]
-        [InlineData(new[] { "a-e:a", "a-s:a", "r-s:a", "r-e:a" }, new[] { true, false, false, true })]
-        [InlineData(new[] { "a-s:a", "a-e:b", "r-s:a", "r-e:b" }, new[] { true, true, true, true })]
-
-        public static async Task RunSharedAndExclusiveLock(string[] steps, bool[] results)
+        [Fact]
+        public static void WritersAreMutuallyExclusiveFromReaders()
         {
-            Debug.Assert(steps != null);
-            Debug.Assert(results != null);
-            Debug.Assert(steps.Length == results.Length);
-            var guid = Guid.NewGuid().ToString();
-
-            int i = 0;
-            var acquirers = new Dictionary<string, List<string>>();
+            var name = Guid.NewGuid().ToString();
+            var read = false;
 
-            foreach (var step in steps)
+            using (var barrier = new Barrier(2))
             {
-                await Task.Yield();
-                var parts = step.Split(new[] { ':' });
-                Debug.Assert(parts.Length == 2);
-                string acquirer = null;
-                bool acquired = false;
-                switch (parts[0])
-                {
-                    case "a-s": // acquire shared lock
-                        (acquired, acquirer) = ProcessUtility.AcquireSharedLock(guid + parts[1]);
-                        Debug.Assert(acquired == results[i], $"{i}");
-                        if (acquired)
+                Task.WaitAll(
+                    Task.Run(() =>
+                    {
+                        using (InterProcessReaderWriterLock.AcquireWriterLock(name))
                         {
-                            var key = "s" + parts[1];
-                            if (!acquirers.TryGetValue(key, out var list))
-                            {
-                                acquirers[key] = list = new List<string>();
-                            }
-                            list.Add(acquirer);
+                            barrier.SignalAndWait();
+                            Thread.Sleep(500);
+                            Assert.False(read);
                         }
-                        break;
-                    case "a-e": // acquire exclusive lock
-                        (acquired, acquirer) = ProcessUtility.AcquireExclusiveLock(guid + parts[1]);
-                        Debug.Assert(acquired == results[i], $"{i}");
-                        if (acquired)
+                    }),
+                    Task.Run(() =>
+                    {
+                        barrier.SignalAndWait();
+                        using (InterProcessReaderWriterLock.AcquireWriterLock(name))
```

Comment:
From the test name, I guess it should be `InterProcessReaderWriterLock.AcquireReaderLock` here?

Reply:
I followed the old rule, all test related to the process in here, like InterProcessMutex